src/r5py/r5/custom_cost_transport_network.py

```diff
@@ -0,0 +1,243 @@
+#!/usr/bin/env python3
+
+"""Subclass for TransportNetwork, enables custom cost routing."""
+import com.conveyal.r5
+from r5py.r5.transport_network import TransportNetwork
+from r5py.util.custom_cost_conversions import (
+    convert_java_hashmap_to_python_dict,
+    convert_python_custom_costs_to_java_custom_costs,
+)
+from r5py.util.exceptions import CustomCostDataError
+from r5py.util.jvm import start_jvm
+
+__all__ = ["CustomCostTransportNetwork"]
+
+start_jvm()
+
+
+def r5_supports_custom_costs():
+    """
+    Check if the R5 java has the GP2 (Green Paths 2) patch i.e. supports custom costs routing.
+
+    Returns:
+    --------
+    bool: True if using GP2 R5, False otherwise.
+    """
+    try:
+        import com.conveyal.r5.rastercost.CustomCostField  # noqa: F401
+
+        # the import was successful thus using GP2 R5
+        return True
+    except ImportError:
+        # the import was unsuccessful
+        return False
+
+
+class CustomCostTransportNetwork(TransportNetwork):
+    """Inherit from TransportNetwork, adds custom cost data routing functionality."""
+
+    def __init__(self, osm_pbf, names, sensitivities, custom_cost_datas):
```

Comment:
data does not have a plural (yes, this is nitpicking, feel free to ignore), you could use ‘data_sets’ instead

Reply:
good point, so will change name to e.g. "name_data_sets"

Reply:
oh you probably meant e.g. "data_names" rather than "name_data_sets"? I think that data_names" will be better anyhow.

Reply:
so actually going to change only the custom_cost_datas which was probably what you meant. Just finished my coffee...